tls/s2n_connection.h

```diff
@@ -304,3 +304,6 @@ extern int s2n_connection_get_protocol_preferences(struct s2n_connection *conn,
 extern int s2n_connection_set_client_auth_type(struct s2n_connection *conn, s2n_cert_auth_type cert_auth_type);
 extern int s2n_connection_get_client_auth_type(struct s2n_connection *conn, s2n_cert_auth_type *client_cert_auth_type);
 extern int s2n_connection_get_client_cert_chain(struct s2n_connection *conn, uint8_t **der_cert_chain_out, uint32_t *cert_chain_len);
+
+int s2n_connection_clear_all_key_shares(struct s2n_connection *conn);
+int s2n_connection_add_preferred_key_share(struct s2n_connection *conn, const char *curve_name);
```

Comment:
Adding key_share by iana_id as done in `s2n_connection_add_preferred_key_share_by_group` seems sufficient.  We probably don't need this function as the only difference here is adding keyshare by curve name.

Reply:
Makes sense, will reduce scope.